test/e2e/http2_disable_test.go

```diff
@@ -0,0 +1,245 @@
+	}
+	if err := waitForRouterDeploymentHTTP2Disabled(kclient, routerDeployTimeout, ic, false); err != nil {
+		t.Fatalf("expected router deployment to have http/2 enabled: %v", err)
+	}
+
+	//
+	// Now assert that changing the "cluster" ingress
+	// configuration will also disable/enable http/2 for all
+	// ingress controllers.
+
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, ingressConfig, routerDeployment, err = http2RefreshTestObjects(t, kclient, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get test objects: %v", err)
+	}
+	if err := checkHTTP2IsEnabled(ic, ingressConfig, routerDeployment); err != nil {
+		t.Fatalf("test assertions failed: %v", err)
+	}
+
+	// Disable http/2 on the ingress config "cluster"
+	if err := disableHTTP2ForIngressConfig(t, kclient, 1*time.Minute, true, ingressConfig); err != nil {
+		t.Fatalf("failed to update infrastructure config: %v", err)
+	}
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+	if err := waitForRouterDeploymentHTTP2Disabled(kclient, routerDeployTimeout, ic, true); err != nil {
+		t.Fatalf("expected router deployment to have http/2 disabled: %v", err)
+	}
+
+	// Revert the previous change to the ingress config
+	ic, ingressConfig, routerDeployment, err = http2RefreshTestObjects(t, kclient, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get test objects: %v", err)
+	}
+	if err := disableHTTP2ForIngressConfig(t, kclient, 1*time.Minute, false, ingressConfig); err != nil {
+		t.Fatalf("failed to update infrastructure config: %v", err)
+	}
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+	if err := waitForRouterDeploymentHTTP2Disabled(kclient, routerDeployTimeout, ic, false); err != nil {
+		t.Fatalf("expected router deployment to have http/2 enabled: %v", err)
+	}
+
+	// Assert all our mutations are back to their initial state
+	ic, ingressConfig, routerDeployment, err = http2RefreshTestObjects(t, kclient, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get test objects: %v", err)
+	}
+	if err := checkHTTP2IsEnabled(ic, ingressConfig, routerDeployment); err != nil {
+		t.Fatalf("test assertions failed: %v", err)
+	}
+}
+
+func http2IsDisabledInEnv(env []corev1.EnvVar) bool {
+	for _, v := range env {
+		if v.Name == ingresscontroller.RouterDisableHTTP2EnvName {
+			if val, err := strconv.ParseBool(v.Value); err == nil {
+				return val
+			}
+		}
+	}
+	return false
+}
+
+func http2RefreshTestObjects(t *testing.T, client client.Client, timeout time.Duration) (*operatorv1.IngressController, *configv1.Ingress, *appsv1.Deployment, error) {
```

Comment:
Looks like `t.Helper()` got lost.

Reply:
These functions no longer call `Fatal` - they return a wrapped error. We carry `t` for logging through `t.Logf()`.